sklearn/decomposition/nmf.py

```diff
@@ -545,6 +545,12 @@ def _multiplicative_update_w(X, W, H, beta_loss, l1_reg_W, l2_reg_W, gamma,
 
         if beta_loss == 1:
             np.divide(X_data, WH_safe_X_data, out=WH_safe_X_data)
+        elif beta_loss == 0:
+            # using numpy's reciprocal function for exponent -1
```

Comment:
If you're effectively using np.reciprocal  and np.square, you could just do that here...